online-shop/src/components/Details.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { RouteComponentProps} from 'react-router-dom';
+import products from './productArray';
+import '../App.css';
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';
+import Button from '@material-ui/core/Button';
+
+type TParams = { id: string };
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    margin: {
+       margin: theme.spacing(1),
+    },
+    button: {
+        margin: theme.spacing(5),
+        fontSize: 25
+    },
+   }),
+);
+
+function Details({match}: RouteComponentProps<TParams>) {
```

Comment:
Please rename it to ProductDetails, to have a better naming convention and a clearer overview of what this component displays